grudge/discretization.py

```diff
@@ -519,6 +521,38 @@ def order(self):
         from pytools import single_valued
         return single_valued(egrp.order for egrp in self._volume_discr.groups)
 
+    # {{{ Discretization-specific geometric properties
+
+    def nodes(self, dd=None):
+        r"""Return the nodes of a discretization.
+
+        :arg dd: a :class:`~grudge.dof_desc.DOFDesc`, or a value convertible to one.
+            Defaults to the base volume discretization.
+        :returns: an object array of :class:`~meshmode.dof_array.DOFArray`\ s
+        """
+        if dd is None:
+            return self._volume_discr.nodes()
+        else:
+            return self.discr_from_dd(dd).nodes()
+
+
+    @memoize_method
+    def normal(self, dd):
```

Comment:
Will the memoization here do anything to prevent update if the mesh moves?  Just curious.

Reply:
That's a good question. I don't think so. This would have to be inferred I guess if the `nodes` are changing.

Reply:
See https://github.com/inducer/meshmode/issues/123 for some related discussion. IMO, generally the safest approach would be to discard and rebuild the whole collection when it changes, while working to make that a (reasonably) cheap operation.

Reply:
Will this discarding and rebuilding be ok when the structures surface regresses at every substep?  Even in the absence of regression (caused by erosion, or burning); if we have real coupling there will be deformation displacements - which definitely occur at the substep level; that is, every time the surface load (i.e. the fluid pressure) changes.  This approach seems like a lot of tear-down-setup overhead.

Reply:
The thing is that making new objects is (relatively) cheap. If it's a new mesh, then by far the most expensive thing will be remapping to a new mesh. If it's smooth deformations, then IMO the cost will be dominated by updating/recomputing area elements and mapping derivatives. Making a new `DiscretizationCollection`, by comparison, is dirt cheap.

Reply:
[That constructor](https://github.com/inducer/grudge/blob/0a7b2fb2e4e170809b55e0c3c193ae1dbc4a61a4/grudge/discretization.py#L74-L162) currently does a bit more work than it needs to, but I think that can be addressed. #104